sklearn/metrics/_regression.py

```diff
@@ -137,6 +140,63 @@ def _check_reg_targets(y_true, y_pred, multioutput, dtype="numeric", xp=None):
     return y_type, y_true, y_pred, multioutput
 
 
+def _check_reg_targets_with_floating_dtype(
+    y_true, y_pred, sample_weight, multioutput, xp=None
+):
+    """Ensure that y_true and y_pred correspond to the same regression task.
+
+    This helper automatically selects a floating-point data type with
+    `_find_matching_floating_dtype` when computing with inputs that
+    follow the array API specification and ensure that all non-None inputs
+    are converted to that common dtype if needed before returning them.
```

Comment:
```suggestion
    Works like `_check_reg_targets` but determines a suitable floating-point data
    type automatically. Only use this with array API compatible inputs.
```

What do you think of this? I think we can say that this works like `_check_reg_targets` as a short cut?

I am not sure how to best include the "only works with array API inputs" - though I am not even sure if this is true. But I think we only want people to use this for array API inputs. We should somehow clarify that and/or make it raise when used in the wrong place?

Reply:
```suggestion
    Ensures that y_true, y_pred, and sample_weight correspond to the same
    regression task.

    Extends `_check_reg_targets` by automatically selecting a suitable floating-point
    data type for inputs using `_find_matching_floating_dtype`.

    Use this private method only when converting inputs to array API-compatibles.
```

WDYT?